### DISCUSSION TITLE
How do I obtain the attention, query and key of fine tuned model output?

### DISCUSSION THREAD

**Discussion by yyyyawend** [NVIDIA-NeMo/NeMo#5245] [Q&A] [answered]
Hello all,

I am trying to visualise attention throughout the fine-tuned BioMegatron Model with token classification(ner) and text classification(re) downstream tasks using bertviz. 
https://colab.research.google.com/drive/1hXIQ77A4TYS4y3UthWF-Ci7V7vVUoxmQ?usp=sharing#scrollTo=YLAhBxDSScmV

How can I get the attention, query, and key sequences? It seems that the NeMo code cannot be edited on Colab.

Can anyone assist?

Thanks,

**Comment by okuchaiev** [comment, marked as answer]
We don't expose this to the user. But NeMo's models are just Pytorch models and code is open-sourced, so you can attempt to expose what you need yourself.
